packages/flutter_tools/lib/src/asset.dart

```diff
@@ -203,6 +230,24 @@ class _Asset {
 
   @override
   String toString() => 'asset: $assetEntry';
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other)) return true;
```

Comment:
Looking at other Flutter implementations of `==` it seems the convention is to do:
```dart
@override
bool operator ==(dynamic other) {
  if (identical(other, this))
    return true;
  if (other.runtimeType != runtimeType)
    return false;
  final _Asset typedOther = other;
  return typedOther.base == base
      && typedOther.assetEntry == assetEntry
      && typedOther.relativePath == relativePath
      && typedOther.source == source;
}
```